aesara/tensor/basic.py

```diff
@@ -1302,17 +1302,26 @@ def eye(n, m=None, k=0, dtype=None):
 
     Returns
     -------
-    ndarray of shape (N,M)
-        An array where all elements are equal to zero, except for the `k`-th
-        diagonal, whose values are equal to one.
-
+    aesara tensor of shape (N,M)
+        A symbolic tensor representing a matrix where all elements are equal to zero,
+        except for the `k`-th diagonal, whose values are equal to one.
```

Comment:
We've been replacing docstring type hints with type annotations, so you can just add the shape information (i.e. `(N, M)`) to the description block and add the type hints to the function signature.  In this case, the return value is only going to be a `TensorVariable`, at least until larger typing efforts like https://github.com/aesara-devs/aesara/pull/1207 go through.